core-web/libs/portlets/dot-ema/src/lib/feature/dot-ema.component.ts

```diff
@@ -106,4 +170,65 @@ export class DotEmaComponent implements OnInit {
             queryParamsHandling: 'merge'
         });
     }
+
+    /**
+     * Create the url to edit a contentlet
+     *
+     * @private
+     * @param {string} inode
+     * @return {*}
+     * @memberof DotEmaComponent
+     */
+    private createEditContentletUrl(inode: string): string {
+        return `/c/portal/layout?p_p_id=content&p_p_action=1&p_p_state=maximized&p_p_mode=view&_content_struts_action=%2Fext%2Fcontentlet%2Fedit_contentlet&_content_cmd=edit&inode=${inode}`;
+    }
+
+    /**
+     * Handle the custom events from the iframe
+     *
+     * @private
+     * @param {Event} event
+     * @memberof DotEmaComponent
+     */
+    private handleNgEvent(event: CustomEvent) {
+        const { detail } = event;
+
+        // Skip the loaded event
+        if (detail.name === CUSTOM_EVENTS.EDIT_CONTENTLET_LOADED) return;
+
+        // This forces a reload in the iframe
+        this.iframe.nativeElement.contentWindow?.postMessage('ema-reload-page', this.host);
+    }
+
+    /**
+     * Handle the post message event
+     *
+     * @private
+     * @param {{ action: MESSAGE_ACTIONS; payload: DotCMSContentlet }} data
+     * @return {*}
+     * @memberof DotEmaComponent
+     */
+    private handlePostMessage({
+        origin = this.host,
+        data
+    }: {
+        origin: string;
+        data: { action: MESSAGE_ACTIONS; payload: DotCMSContentlet };
+    }): () => void {
+        const action = origin !== this.host ? 'NOOP' : data.action;
+
+        return {
+            [MESSAGE_ACTIONS.EDIT_CONTENTLET]: () => {
+                this.loadingIframe = true;
+                this.visible = true;
+                this.header = data.payload.title;
+                this.dialogIframe.nativeElement.src = this.createEditContentletUrl(
+                    data.payload.inode
+                );
```

Comment:
Can't we just have a `<iframe [src]="editIframeUrl" />`

```suggestion
                this.editIframeUrl = this.createEditContentletUrl(
                    data.payload.inode
                );
```

Reply:
Also all this is UI state change... didn't we talked about having this in the component store?

Reply:
Moved to the store!